### PR TITLE
[image][android] Add abortPrefetch method

### PR DESCRIPTION
**Pull request by mstach60161**
# Why

To enable stop caching process.

# How

- Added abortPrefetch method.

# Test Plan

- Slowed down the network.
- Started prefetching the image.
- Executed `abortPrefetch` function

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).

### CODE REVIEW

**Review by barthap — COMMENTED** [review]
I'm curious - what kind of error does `Image.prefetch()` return when it's aborted? I think you can check it this way:
```ts
Image.prefetch(url, id => {
  Image.abortPrefetch(id);
}).then(console.log)
  .catch(console.log)
```

File: packages/expo-image/src/Image.tsx [hunk 1]
```diff
@@ -69,14 +79,35 @@ export default class Image extends React.Component<ImageProps, ImageState> {
...
+   *
+   * @param callback The function that will be called with the `requestId`. Callback is performed before the prefetch starts.
+   * You can use `abortPrefetch` only after prefetching has started.
+   *
    * @return an empty promise.
    */
-  static async prefetch(url: string): Promise<void> {
-    if (Platform.OS !== 'android') {
+  static async prefetch(url: string, callback?: ImagePrefetchCallback): Promise<void> {
+    if (!ExpoImageModule.abortPrefetch) {
```
Comment (barthap) [inline packages/expo-image/src/Image.tsx:92]:
```suggestion
    if (Platform.OS !== 'android') {
```
Perhaps you accidentally overwritten the correct `if`

Comment (mstach60161) [reply packages/expo-image/src/Image.tsx:92]: No, I would like to keep the same convention in both methods. What do you think about it?

Comment (barthap) [reply packages/expo-image/src/Image.tsx:92]: No problem, but then change `ExpoImageModule.abortPrefetch` to `ExpoImageModule.prefetch` 😉

Comment (mstach60161) [reply packages/expo-image/src/Image.tsx:92]: Oh, yes, I missed it.

**Comment by ide** [comment]
Since this is making an API design change the Expo team needs to review and approve the proposal. Before going further with the code, I'd like to see a proposed API and explanations for how this would be implemented on Android, iOS, and the web.

Instead of adding a new method, could you please propose an API that uses the W3C standard AbortController with `prefetch`?

**Comment by bbarthec** [comment]
As we're aiming to achieve API identity between our component and the original component from `react-native` we need to adjust the current `abortPrefetch` method according to https://github.com/expo/expo/discussions/14351.
As this additional parameter exists only on Android platform in the original `Image` component from `react-native` I'm not sure if we need Web/iOS proposal at this point 🤔 I believe filling the gaps in the original `Image` API and closing the existing gaps between the platforms is the subject of later iterations over this package.

**Comment by ide** [comment]
@bbarthec sent you a message, I'd like the base API to be stable and reliable, and we can provide compatibility layers separately without affecting the base API too much.

**Pull request closed without merging**